cmake/hip.cmake

```
@@ -212,9 +212,13 @@ if(GINKGO_HIP_AMDGPU)
     endforeach()
 endif()
 
+
 set(GINKGO_HIPCC_OPTIONS ${GINKGO_HIP_COMPILER_FLAGS} "-std=c++14 -DGKO_COMPILING_HIP")
 set(GINKGO_HIP_NVCC_OPTIONS ${GINKGO_HIP_NVCC_COMPILER_FLAGS} ${GINKGO_HIP_NVCC_ARCH} ${GINKGO_HIP_NVCC_ADDITIONAL_FLAGS})
 set(GINKGO_HIP_CLANG_OPTIONS ${GINKGO_HIP_CLANG_COMPILER_FLAGS} ${GINKGO_AMD_ARCH_FLAGS})
+if(GINKGO_HIP_AMD_UNSAFE_ATOMIC AND HIP_VERSION VERSION_GREATER_EQUAL 5)
+set(GINKGO_HIP_CLANG_OPTIONS ${GINKGO_HIP_CLANG_OPTIONS} "-munsafe-fp-atomics")
```

nit: indentation + CMake stuff
```suggestion
    list(APPEND GINKGO_HIP_CLANG_OPTIONS -munsafe-fp-atomics)
```